instrumentation/net/http/otelhttp/test/semantic_convention_test.go

```diff
@@ -0,0 +1,158 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package test
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
+
+	"go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
+	"go.opentelemetry.io/otel/sdk/resource"
+	"go.opentelemetry.io/otel/sdk/trace"
+)
+
+type lc struct {
+	t testing.TB
+}
+
+func (l *lc) Accept(log testcontainers.Log) {
+	l.t.Log(string(log.Content))
+}
+
+var docker = flag.Bool("docker", false, "Enable tests that use docker.")
```

Comment:
Should there be a specific test suite running this in CI? Or do you plan this for later?
Also, an alternative here rather than use the `flag` package could be to use a `// go:build docker` build flag so the file doesn't run by default, but can be enabled.

Reply:
> // go:build docker

Let's not do that. Many tools (IDE, golangci-lint) might stop working.

However, I think we could (should) enable it by default.

We can consider skipping when `-short` is provided using this pattern: https://pkg.go.dev/testing#hdr-Skipping.

Reply:
I made it part of the non-`-short` tests.  I do want to warn that this makes docker required for running all tests.  including `make test` and `make precommit`